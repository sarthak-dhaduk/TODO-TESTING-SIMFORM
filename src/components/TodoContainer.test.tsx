import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import TodoContainer from "./TodoContainer";

describe("TodoContainer component", () => {
  //testing the ui snapshot
  test("TodoContainer renders correctly", () => {
    const renderedComponent = renderer.create(<TodoContainer />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });
});
