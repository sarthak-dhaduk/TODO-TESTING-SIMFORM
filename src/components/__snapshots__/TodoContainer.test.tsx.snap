// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoContainer component TodoContainer renders correctly 1`] = `
<div
  className="h-screen flex flex-col justify-center items-center text-textLight"
>
  <div
    className="static rounded overflow-hidden shadow-lg bg-white w-4/5 sm:w-1/2 lg:w-2/5 xl:w-1/3 2xl:w-1/4 px-10 h-[75vh] flex flex-col"
  >
    <div
      className="flex justify-between my-4 mt-12 items-center"
    >
      <div
        className="flex"
      >
        <div
          className="text-5xl mr-3 font-medium"
        >
          19
        </div>
        <div
          className="flex flex-col"
        >
          <div
            className="font-medium"
          >
            APRIL
          </div>
          2023
        </div>
      </div>
      <div
        className="font-medium"
      >
        WEDNESDAY
      </div>
    </div>
    <div
      className="h-[100%] overflow-auto "
    >
      <ul>
        <li
          className="flex justify-between items-center mt-8"
        >
          <div
            className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg text-textComplete"
          >
            Buy new sweatshirt
          </div>
          <button
            className="w-9"
          >
            <img
              alt="completed"
              src="/assets/task-completed.png"
            />
          </button>
        </li>
        <li
          className="flex justify-between items-center mt-8"
        >
          <div
            className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg text-textComplete"
          >
            Begin promotional phase
          </div>
          <button
            className="w-9"
          >
            <img
              alt="completed"
              src="/assets/task-completed.png"
            />
          </button>
        </li>
      </ul>
      <ul>
        <li
          className="flex justify-between items-center mt-8"
        >
          <div
            className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
          >
            Read an article
          </div>
          <button
            className="w-9"
          >
            <img
              alt="completed"
              src="/assets/task-pending.png"
            />
          </button>
        </li>
        <li
          className="flex justify-between items-center mt-8"
        >
          <div
            className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
          >
            Try not to sleep
          </div>
          <button
            className="w-9"
          >
            <img
              alt="completed"
              src="/assets/task-pending.png"
            />
          </button>
        </li>
        <li
          className="flex justify-between items-center mt-8"
        >
          <div
            className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
          >
            Watch 'Sherlock'
          </div>
          <button
            className="w-9"
          >
            <img
              alt="completed"
              src="/assets/task-pending.png"
            />
          </button>
        </li>
        <li
          className="flex justify-between items-center mt-8"
        >
          <div
            className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
          >
            Begin QA for the product
          </div>
          <button
            className="w-9"
          >
            <img
              alt="completed"
              src="/assets/task-pending.png"
            />
          </button>
        </li>
        <li
          className="flex justify-between items-center mt-8"
        >
          <div
            className="break-words text-justify flex-1 mx-2 max-w-[80%] text-lg "
          >
            Go for a walk
          </div>
          <button
            className="w-9"
          >
            <img
              alt="completed"
              src="/assets/task-pending.png"
            />
          </button>
        </li>
      </ul>
    </div>
    <button
      className="absolute bottom-[calc(11vh-30px)] left-[calc(50vw-38px)] rounded-full bg-lightGreen p-7 opacity-95"
      onClick={[Function]}
    >
      <img
        alt="plus"
        className="opacity-40 w-6"
        src="/assets/plus.png"
      />
    </button>
  </div>
</div>
`;
